app/src/main/java/de/westnordost/streetcomplete/data/osmtracks/TracksApiImpl.kt

```diff
@@ -0,0 +1,82 @@
+package de.westnordost.streetcomplete.data.osmtracks
+
+import de.westnordost.osmapi.OsmConnection
+import de.westnordost.osmapi.common.errors.OsmApiException
+import de.westnordost.osmapi.common.errors.OsmApiReadResponseException
+import de.westnordost.osmapi.common.errors.OsmAuthorizationException
+import de.westnordost.osmapi.common.errors.OsmConflictException
+import de.westnordost.osmapi.common.errors.OsmConnectionException
+import de.westnordost.osmapi.common.errors.OsmQueryTooBigException
+import de.westnordost.osmapi.map.data.OsmLatLon
+import de.westnordost.osmapi.traces.GpsTraceDetails
+import de.westnordost.osmapi.traces.GpsTracesApi
+import de.westnordost.osmapi.traces.GpsTrackpoint
+import de.westnordost.streetcomplete.ApplicationConstants
+import de.westnordost.streetcomplete.data.download.ConnectionException
+import de.westnordost.streetcomplete.data.download.QueryTooBigException
+import de.westnordost.streetcomplete.data.upload.ConflictException
+import de.westnordost.streetcomplete.data.user.AuthorizationException
+import java.time.Instant
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+
+class TracksApiImpl(osm: OsmConnection) : TracksApi {
+    private val api: GpsTracesApi = GpsTracesApi(osm)
+
+    override fun create(tracks: List<Trackpoint>): Track = wrapExceptions {
+
+        // Filename is just the start of the track
+        // https://stackoverflow.com/a/49862573/7718197
+        val name = DateTimeFormatter
+            .ofPattern("yyyy-MM-dd HH:mm:ss.SSSSSS")
+            .withZone(ZoneOffset.UTC)
+            .format(Instant.ofEpochSecond(tracks[0].time))
+            .replace("-", "_")
+            .replace(":", "_")
+            .replace(" ", "_") + ".gpx"
+        val visibility = GpsTraceDetails.Visibility.IDENTIFIABLE
+        val description = ApplicationConstants.USER_AGENT
+        val tags = listOf(ApplicationConstants.NAME.lowercase())
+
+        // Generate history of trackpoints
+        val trackpoints = mutableListOf<GpsTrackpoint>()
+        tracks.forEachIndexed { idx, it ->
+            trackpoints.add(
+                GpsTrackpoint(
+                    OsmLatLon(it.position.latitude, it.position.longitude),
+                    Instant.ofEpochMilli(it.time),
+                    idx == 0,
+                    it.horizontalDilutionOfPrecision,
+                    it.elevation
+                )
+            )
+        }
```

Comment:
```suggestion
        val trackpoints = tracks.mapIndexed { idx, it -> 
            GpsTrackpoint(
                OsmLatLon(it.position.latitude, it.position.longitude),
                Instant.ofEpochMilli(it.time),
                idx == 0,
                it.horizontalDilutionOfPrecision,
                it.elevation
            )
        }
```

You can use `mapIndexed`, it's shorter